src/client/pkg/tls/cert_loader.go

```diff
@@ -0,0 +1,79 @@
+package tls
+
+import (
+	"crypto/tls"
+	"fmt"
+	"sync/atomic"
+	"time"
+	"unsafe"
+
+	"github.com/pachyderm/pachyderm/src/client/pkg/errors"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// CertLoader provides simple hot TLS certificate reloading by checking for a renewed certificate at a configurable interval
+type CertLoader struct {
+	certPath        string
+	keyPath         string
+	refreshInterval time.Duration
+
+	// cert is the current cached *tls.Certificate. It should only be accessed with atomic methods because it may be updated by the cert reloading routine.
+	cert     unsafe.Pointer
+	stopChan chan interface{}
+}
+
+func NewCertLoader(certPath, keyPath string, refreshInterval time.Duration) *CertLoader {
+	return &CertLoader{
+		certPath:        certPath,
+		keyPath:         keyPath,
+		refreshInterval: refreshInterval,
+	}
+}
+
+// loadAndStart ensures the current TLS certificate is loaded and starts the reload routine to poll for renewed certificates
+func (l *CertLoader) LoadAndStart() error {
+	if err := l.loadCertificate(); err != nil {
+		return err
+	}
+	go l.reloadRoutine()
+	return nil
+}
+
+// stop signals the reloading routine to stop
+func (l *CertLoader) Stop() {
```

Comment:
Small paranoid thing here, do you think it makes sense to have a `stopped` field on the `CertLoader` and add:

```
if stopped {
    return
}
stopped = true
```

Just in case someone in the future double calls `Stop` we can avoid the panic.